app/components/concept-admin/delete-concept-modal.ts

```diff
@@ -0,0 +1,36 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import type RouterService from '@ember/routing/router-service';
+import type Store from '@ember-data/store';
+import type ConceptModel from 'codecrafters-frontend/models/concept';
+
+type Signature = {
+  Element: HTMLDivElement;
+
+  Args: {
+    onClose?: () => void;
+    concept: ConceptModel;
+  };
+};
+
+export default class DeleteConceptModalComponent extends Component<Signature> {
+  @service router!: RouterService;
+  @service store!: Store;
+
+  @action
+  async deleteConcept() {
+    if (!this.args.concept) {
```

Comment:
When will this happen? The type for `concept` suggests that it'll always be `ConceptModel`.